be/src/olap/rowid_conversion.cpp

```diff
@@ -0,0 +1,161 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/rowid_conversion.h"
+namespace doris {
+RowIdConversion::RowIdConversion() = default;
+RowIdConversion::~RowIdConversion() {
+    if (_file_reader.get() != nullptr) {
+        auto st = _file_reader->close();
+        if (!st.ok()) {
+            LOG(WARNING) << "fail to close file " << _file_name;
+        }
+        st = io::global_local_filesystem()->delete_file(_file_name);
```

Comment:
If the deletion fails, will there be some problems creating a file with the same name in the next compaction？

Reply:
if delete fail or be crash, this file will exist on disk, but when next time doing this tablet compaction, it will delete the existing file first. There is only one base or cumulative comaction task is doing on the sepcific tablet, and i have make file name end with compaction type, so we don't need to worry about file conflict between base and  cumulative comaction  on one tablet.